bin/hdfcoinc/pycbc_coinc_statmap

```diff
@@ -133,20 +211,269 @@ logging.info("calculating ifar/fap values")
+                     "--max-hier-removal too low!")
+        break
+
+    # We're doing a hierarchical removal so add the iteration number to the
+    # next one. Parse that to a string for later use too.
+    h_iterations += 1
+    h_iter_str = str(h_iterations)
+
+    # Find the loudest of these triggers, its index, its newsnr
+    # We know that the IFARs of these triggers are all louder
+    # than the inclusive background
+    max_ifar_ind = loud_t_ind[0]
+    for i in range(0, len(loud_t_ind)):
+        # Check to see which of these louder than all of the background
+        # foreground triggers has the greatest snr. We will assign this
+        # location the "max_ifar_ind" and remove that trigger first.
+        if fore_stat[loud_t_ind[i]] > fore_stat[max_ifar_ind]:
+            max_ifar_ind = loud_t_ind[i]
+
+    # Step 3: Remove that trigger from the list of zerolag triggers
+
+    # Store where the loud foreground trigger that I want to remove next in the
+    # list of all of the triggers located is at
+    where_is_fore_t = numpy.where(all_trigs.stat[:] == fore_stat[max_ifar_ind])[0]
+
+    # FIXME:
+    # In the RARE case that there are two equally loud
+    # louder-than-the-background foreground triggers give preference
+    # to the one with the earliest GPS time, just for documenting purposes
+    if len(where_is_fore_t) > 1:
+        logging.warn("WARNING: There is more than one foreground trigger "
+                     " that is louder than the background and have the same "
+                     "ranking statistic. This code may operate strangely in"
+                     "this condition.")
+
+    # This is the foreground trigger index in the all_trigs array that we
+    # want to remove
+    my_loudest_ind = where_is_fore_t[0]
+
+    # Store any foreground trigger's information that we want to
+    # hierarchically remove. Below save the ifar.
+
+    save_ifar_fore_trig = numpy.append(save_ifar_fore_trig, ifar[max_ifar_ind])
+    # The following equation should be safe bcause coinc_time
+    # will only be reassigned after we remove this trigger for the
+    # next quietest louder-than-the-inclusive-background foreground
+    # trigger.
+    save_fap_fore_trig = numpy.append(save_fap_fore_trig, \
+                                   1 - numpy.exp(- coinc_time / ifar[max_ifar_ind]))
+
+    # Save the location in the foreground trigger array too
+    save_these_fore_ind = numpy.append(save_these_fore_ind, max_ifar_ind)
+
+    logging.info("Removing foreground trigger that is louder than the inclusive background.")
+
+    # Remove this foreground trigger from the list of loud foreground triggers
+    loud_t_ind = numpy.delete(loud_t_ind, numpy.where(loud_t_ind[:] == max_ifar_ind)[0][0])
+
+    # Remove the foreground trigger AND all of the inclusive background
+    # triggers that are associated with it via the timeslides.
+
+    ind_to_rm_ifo1 = veto.indices_within_times(all_trigs.time1, \
```

Comment:
Also, to be consistent, calculate a single average time for the trigger, and then use that here.